src/pages/schedule.js

```diff
@@ -0,0 +1,157 @@
+    if (state.user && !form.maxBooked) {
+      fetch("http://localhost:5000/getUserEvents", {method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ uid: state.user.uid, firstName: state.user.firstName, lastName: state.user.lastName })})
+        .then((res) => res.json())
+        .then((res) => {
+          setForm({ ...form, isLoading: false })
+
+          let calendarElement = document.getElementById("calendarElement")
+
+          // initialize the Calendar object to be rendered to the DOMs
+          let calendar = new Calendar(calendarElement, {plugins: [dayGridPlugin, googleCalendarPlugin, interactionPlugin],
+            initialView: "dayGridMonth",
+
+            eventMouseover: function () {},
+            eventClick: function (info) {
+              // do not allow the booking of a past event
+              const currentTime = new Date()
+              const eventTime = new Date(info.event.start)
+
+              if (eventTime < currentTime) {
+                setForm({...form,
+                  isLoading: false,
+                  message: { text: "Error in booking lesson - you cannot book a lesson in the past", type: "error" }})
+
+                return
+              }
+
+              if (info.event.extendedProps.booked === true) {
+                setForm({...form,
+                  isLoading: false,
+                  message: { text: "Error in booking lesson - this lesson has already been booked", type: "error" }})
+
+                return
+              }
+
+              // ensure that the user hasn't already booked their max amount of lessons
+              if (form.maxBooked === false) {
+                const lessonTime = info.event.start
+                const formattedLessonTime = format(lessonTime, "EEEE, LLLL do, y 'at' h:mm a '(PST)'")
+                setForm({...form,
+                  isLoading: false,
+                  targetLessonId: info.event.id,
+                  message: { text: "Are you sure you want to book this time: " + formattedLessonTime, type: "confirm" }})
+              } else {
+                setForm({...form,
+                  isLoading: false,
+                  targetLessonId: "",
+                  message: {text:
+                      "You have already booked a lesson, if you would like to change your lesson please email treetoplearningorg@gmail.com",
+                    type: "success"}})
+              }
+            },
+            weekends: false,
+            header: false,
+            events: res.events})
+          calendar.render()
+        })
+        .catch((err) => console.log("error in fetching events", err))
+    }
+  }, [state.user, form.isLoading])
+
+  return (
+    <div className="w-screen min-h-screen bg-base">
+      <Navbar />
+      {form.isLoading && (
+        <div className="flex self-center justify-center w-screen h-screen ">
+          {" "}
+          <LoadingAnimation data={gear} />
```

Comment:
Honestly should be `LoadingSpinner` and all of this stuff in the parent could easily just be tossed into the component (that way it adapts to its parent all the time). Will reduce ~4 lines of code per page.